File: trunk/src/Servidor/Transaccional/AccesoDatos/Dao/NHibernate/DaoRolNHibernate.cs
﻿using System;
using System.Collections.Generic;
using NHibernate;
using NLog;
using Trascend.Bolet.AccesoDatos.Contrato;
using Trascend.Bolet.ObjetosComunes.Entidades;
using System.Configuration;

namespace Trascend.Bolet.AccesoDatos.Dao.NHibernate
{
    public class DaoRolNHibernate : DaoBaseNHibernate<Rol, string>, IDaoRol
    {
        private static Logger logger = LogManager.GetCurrentClassLogger();


        public IList<Rol> ObteneRolesYObjetos()
        {
            IList<Rol> roles;

            try
            {

                #region trace
                if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                    logger.Debug("Entrando al Método {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
                #endregion

                IQuery query = Session.CreateQuery(Recursos.ConsultasHQL.ObtenerRolesYObjetos);
                roles = query.List<Rol>();

                #region trace
                if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                    logger.Debug("Saliendo del Método {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
                #endregion
            }
            catch (Exception ex)
            {
                logger.Error(ex.Message);
                throw new ApplicationException(Recursos.Errores.ExConsultarTodos);
            }
            finally
            {
                Session.Close();
            }

            return roles;
        }
        
    }
}

File: src/Servidor/Transaccional/AccesoDatos_Fac/Dao/NHibernate/DaoRolNHibernate.vb
Imports System.Collections.Generic
Imports NHibernate
Imports NLog
Imports Diginsoft.Bolet.AccesoDatos.Contrato
Imports Diginsoft.Bolet.ObjetosComunes.Entidades

Namespace Dao.NHibernate
    Public Class DaoRolNHibernate
        Inherits DaoBaseNHibernate(Of Rol, String)
        Implements IDaoRol

        Private Shared logger As Logger = LogManager.GetCurrentClassLogger()

        Public Function ObteneRolesYObjetos() As System.Collections.Generic.IList(Of ObjetosComunes.Entidades.Rol) Implements Contrato.IDaoRol.ObteneRolesYObjetos
            Dim roles As IList(Of Rol)

            Try
                Dim query As IQuery = Session.CreateQuery(Recursos.ConsultasHQL.ObtenerRolesYObjetos)
                roles = query.List(Of Rol)()
            Catch ex As Exception
                logger.[Error](ex.Message)
                Throw New ApplicationException(Recursos.Errores.ExConsultarTodos)
            Finally
                Session.Close()
            End Try

            Return roles
        End Function
    End Class
End Namespace